File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Text:");
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (isalpha(text[i]))
        {
            letters++;
        }
    }

    double L = ((double) letters / words) * 100;
    double S = ((double) sentences / words) * 100;
    double grade = 0.0588 * L - 0.296 * S - 15.8;


    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(grade));
    }
}

File: sentimental-readability/readability.py
def main():
    text = input("Text: ")
    letters = 0
    words = 1
    sentences = 0

    for char in text:
        if char in ['.', '!', '?']:
            sentences += 1
        elif char.isspace():
            words += 1
        elif char.isalpha():
            letters += 1

    L = (letters / words) * 100
    S = (sentences / words) * 100
    grade = 0.0588 * L - 0.296 * S - 15.8

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(grade)}")

if __name__ == "__main__":
    main()
